# Module 2 Lab 1 — Python Warm-Up + First Push

**Name:** Roee Kadur  
**Course:** CS 82A  
**GitHub Repository:** https://github.com/roee0910/cs82a-portfolio

This notebook completes Steps 1–9 of Module 2 Lab 1.


## Step 1: Wages with f-strings

In [1]:
name = "Roee"
hourly_wage = 20.0
hours = 32
weekly_pay = hourly_wage * hours

print(f"{name} earns {weekly_pay} per week")

Roee earns 640.0 per week


## Step 2: Count the warm days

In [2]:
temps = [61, 65, 72, 58, 70, 74, 68]

warm_days = 0
for temp in temps:
    if temp > 65:
        warm_days = warm_days + 1

print(warm_days)

4


## Step 3: Temperature converter

In [3]:
def to_celsius(f):
    return (f - 32) * 5 / 9

print(round(to_celsius(temps[0]), 1))

16.1


## Step 4: A movie dictionary

In [4]:
movie = {
    "title": "The Dark Knight",
    "year": 2008,
    "rating": 9.0
}

for key, value in movie.items():
    print(key, ":", value)

title : The Dark Knight
year : 2008
rating : 9.0


## Step 5: Load and inspect `chickwts.csv`

In [5]:
import pandas as pd

df = pd.read_csv("chickwts.csv", index_col=0)

df.head()

,weight,feed
1,179,horsebean
2,160,horsebean
3,136,horsebean
4,227,horsebean
5,217,horsebean


In [6]:
df.shape

(71, 2)

In [7]:
df.describe()

,weight
count,71.000000
mean,261.309859
std,78.073700
min,108.000000
25%,204.500000
50%,258.000000
75%,323.500000
max,423.000000


## Step 6: Interpret `describe()`

The average chick weight is about **261.3**, while the median is **258**, so a typical chick in this dataset weighs around 260. The standard deviation is about **78.1**, showing a fairly wide spread in weights, with values ranging from **108 to 423**. The large range suggests that the chicks vary substantially in size.

## Step 7: GitHub repository

Create a **PUBLIC** repository named `cs82a-portfolio`.

README text:

> Hi, I'm Roee Kadur. This repository is my portfolio for CS 82A, where I am learning Python and data science foundations.
>
> Module 2: Python foundations


## Step 8: Upload the notebook

Upload this file to the `module2/` folder in the public repository:

`module2/module2_lab.ipynb`


## Step 9: Submit the URL

**GitHub repository URL:**  
https://github.com/roee0910/cs82a-portfolio

Submit `module2_lab.ipynb` and `module2_lab.pdf` individually to Canvas.
